### DISCUSSION TITLE
Relatórios customizáveis para o controle do usuário

### DISCUSSION THREAD

**Discussion by gtkatakura-bysoft** [frontendbr/forum#866] [Perguntas e Respostas]
Alguém conhece alguma ferramenta ou exemplo onde o usuário seja capaz de customizar um relatório existente no sistema, pensando em um simples relatório onde mostra dados gerenciais linha por linha? As customizações de relatório seriam ao nível de:

- Adicionar/remover colunas referente a um registro
- Adicionar agrupadores de registros
- Adicionar totalizadores (seja gerais ou para um agrupador de registro)

Considerando que o usuário tenha conhecimento de HTML, poderia ser disposto o template do relatório para edição do usuário, para controle de quais colunas ele deseja visualizar, o que ele acha interessante agrupar ou totalizar, porém não encontrei algo "pronto", um "esqueleto" para servir de base ao que estamos planejando implementar.

**Comment by renatosramos** [comment]
Olá @gtkatakura-bysoft. 
Recentemente fiz uma pesquisa semelhante a sua, e não encontrei nenhuma ferramente que tivesse uma usabilidade boa para o usuário. As que eu encotrei que talvez possam lhe serem uteis foram essas:

http://webpivottable.com/demo/
https://www.fast-report.com:2015/razor/Home/Designer
https://www.grapecity.com/en/activereports-server#demos

No meu caso eu realmente tive que desenvolver essa ferramenta do zero, porque além de algumas ferramentas serem pagas, também exigiriam do usuário um nível maior de esforço para criar os relatórios. 

Pela minha experiência o que demandará mais tempo é fazer a analise e organização do banco de dados para que seja possível para o usuário selecionar as tabelas e campos que serão exibidos no relatório.

De qualquer forma espero ter ajudado.
